src/bt/Node.cpp

```diff
@@ -114,4 +114,17 @@ std::string Node::status_print(Node::Status s) {
     }
 }
 
+void Node::setRoleString(std::string roleName) {
+    auto children = this->getChildren();
+    if (properties) {
+        this->properties->setString("ROLE", roleName);
+    }
+
+    for (auto child : this->getChildren()) {
+        if (child) {
```

Comment:
can this ever be false? Every child in the vector is a shared pointer that was constructed through make_shared, meaning that .get() != null

Reply:
Children are initialized to nullptr